### ISSUE TITLE
Tests hang on Win32

### ISSUE THREAD

**Issue by gregoa** [libwww-perl/WWW-Mechanize#31]
In case someone is wondering why there are no cpantester reports for 1.75 with Perl 5.22 on Windows: The reason is that the test suite hangs until it is killed by some timeout:

```
Running make test
"C:\strawberry\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-
e" "undef *Test::Harness::Switches; test_harness(0, 'blib\lib', 'blib\arch')" t\
00-load.t t\add_header.t t\aliases.t t\area_link.t t\autocheck.t t\clone.t t\con
tent.t t\cookies.t t\credentials-api.t t\credentials.t t\die.t t\dump.t t\field.
t t\find_frame.t t\find_image.t t\find_inputs.t t\find_link-warnings.t t\find_li
nk.t t\find_link_id.t t\form-parsing.t t\form_with_fields.t t\frames.t t\image-n
ew.t t\image-parse.t t\link-base.t t\link-relative.t t\link.t t\new.t t\pod-cove
rage.t t\pod.t t\regex-error.t t\save_content.t t\select.t t\taint.t t\tick.t t\
untaint.t t\upload.t t\warn.t t\warnings.t t\local\back.t t\local\click.t t\loca
l\click_button.t t\local\content.t t\local\encoding.t t\local\failure.t t\local\
follow.t t\local\form.t t\local\get.t t\local\nonascii.t t\local\overload.t t\lo
cal\page_stack.t t\local\referer.t t\local\reload.t t\local\submit.t t\mech-dump
\mech-dump.t
# Testing WWW::Mechanize 1.75, with LWP 6.13, Perl 5.022000, C:\strawberry\perl\
bin\perl.exe
# Test::Memory::Cycle is not installed.
t\00-load.t .............. ok
t\add_header.t ........... ok
t\aliases.t .............. ok
t\area_link.t ............ ok
t\autocheck.t ............ ok
t\clone.t ................ ok
t\content.t .............. ok
t\cookies.t .............. skipped: HTTP::Server::Simple does not support Window
s yet.
t\credentials-api.t ...... ok
t\credentials.t .......... ok
t\die.t .................. ok
t\dump.t ................. ok
t\field.t ................ ok
t\find_frame.t ........... ok
t\find_image.t ........... ok
t\find_inputs.t .......... ok
t\find_link-warnings.t ... ok
t\find_link.t ............ ok
t\find_link_id.t ......... ok
t\form-parsing.t ......... ok
There are 2 forms with the named fields.  The first one was used. at t\form_with
_fields.t line 27.
t\form_with_fields.t ..... ok
t\frames.t ............... ok
t\image-new.t ............ ok
t\image-parse.t .......... ok
t\link-base.t ............ ok
t\link-relative.t ........ ok
t\link.t ................. ok
Error GETing http://localhost:1473/: Can't connect to localhost:1473 at t\local\
back.t line 53.
```

This makes Mech 1.75 more or less uninstallable on Windows.

**Comment by nponeccop** [comment]
All `t/local/*.t` tests hang if `$^O eq 'MSWin32'` in 1.77 on Windows x64 with Perl 5.24 x64

A possible solution would be to adapt the following snipped from `cookies.t` to all the local tests:
```
if ( $^O =~ /Win32/ ) {
    plan skip_all => 'HTTP::Server::Simple does not support Windows yet.';
}
else {
    plan tests => 14;
}
```

I don't use the module but it's included in StrawberryPerl Perl distribution for Windows, and it fails to update. So probably the guys from Perl-Dist-StrawberryPerl could fix this problem.

**Comment by kenneth-olwing** [comment]
This stumped me when trying to install the 1.83 version, 
Just found what seems to be the problem - I don't believe there's a bug as such, it's because a 'modern' Windows will prefer ipv6, and evidently something inside the Perl code isn't geared up for that properly. I understand that there's supposedly ipv6 support in various places in´, say, Socket, but basically, it doesn't help, at least not here & now. Sorry, digging out the details is a bit beyond me...:-/

Anyway, this is clearly visible by doing 'ping localhost' and it will display pings to the ipv6 '::1' address. But using the URL displayed in the test output directly from the browser shows that the server indeed answers. 

Hence I realized that the Perl code somehow has a problem related to ipv6, and my first attempt was to hack the t/local/LocalServer.pm (line 121) to set the host to '127.0.01' instead of 'localhost' - and that worked fine. That might be a valid solution, at least for the time being as I suppose it'd work on all platforms in all cases (I guess there are few that only have ipv6 at this point).

Another, more 'local to this pc', approach is to use procedures here: https://support.microsoft.com/en-us/help/929852/how-to-disable-ipv6-or-its-components-in-windows, with a wizard to change 'Prefer IPv4 over IPv6 in prefix policies'. Just run the wizard and when complete, reboot. Now a 'ping localhost' shows 127.0.0.1.

Hope this helps.

**Comment by nponeccop** [comment]
Disabling IPv6 system-wide just to safisfy a random retrograde guy seems over the top to me. My advise of just disabling the hanging test is only marginally better though as it doens't include proper fixing either.

**Comment by kenneth-olwing** [comment]
'Prefer IPv4 over IPv6 in prefix policies'

That's the setting I talked about, *NOT* disabling ipv6 altogether (because yes, that would be stupid and not recommended by MS).

And disabling the tests entirely is not necessary either.

I think the simplest fix for the moment is to force ipv4 by using '127.0.0.1' instead of 'localhost' in LocalServer.pm. That should keep the test working everywhere until it's sorted out exactly how ipv6 plays into it and how it should be addressed.

**Comment by karenetheridge** [comment]
I see this too, in version 1.84 on ubuntu linux with perl 5.14.2.

**Comment by karenetheridge** [comment]
If there's no compelling reason to avoid `127.0.0.1` in this test, I think we should make this change. I have this test failing intermittently in a production environment (linux, not windows) and have needed to patch around it.

**Comment by oalders** [comment]
Sounds good to me.  Anybody motivated to create a PR?

**Comment by karenetheridge** [comment]
doing so now.

**Comment by gregoa** [comment]
1.85 still hangs on windows, now with:

```
...
t/local/back.t ........... ok
t/local/click.t .......... ok
t/local/click_button.t ... ok
# Running tests against http://127.0.0.1:49525/?xml=1
t/local/content.t ........ ok
t/local/encoding.t ....... ok
t/local/failure.t ........ ok
t/local/follow.t ......... ok
t/local/form.t ........... ok
t/local/get.t ............ ok
t/local/nonascii.t ....... ok
t/local/overload.t ....... skipped: Mysteriously stopped passing, and I don't know why.
t/local/page_stack.t ..... ok
Error GETing http://localhost:49564/: Can't connect to localhost:49564 at t/local/referer.t line 39.
```

**Comment by karenetheridge** [comment]
oh dear. it did fix me up on ubuntu, though!

**Comment by kenneth-olwing** [comment]
And if you patch that test from 'localhost' is '127.0.0.1' instead?

**Comment by karenetheridge** [comment]
ah, belay that -- the test module that got patched did not change the behaviour of t/local/referer.t, where I still get a 30min hang. We should change t/local/referer-server as well.

**Comment by gregoa** [comment]
Ack, I also just ended up in t/local/referer-server :)
If I change localhost to 127.0.0.1 there, then `gmake test` on this windows box passes.

**Comment by karenetheridge** [comment]
I created another patch, in https://github.com/libwww-perl/WWW-Mechanize/pull/230.

**Comment by oalders** [comment]
Closed vi #230
